cosmwasm/packages/wasmi-runtime/src/lib.rs

```diff
@@ -78,7 +81,50 @@ fn set_log_level_or_default(default: LevelFilter, max_level: LevelFilter) {
         }
     }
 
-    log::set_logger(&LOGGER)
-        .map(|()| log::set_max_level(log_level))
-        .unwrap();
+    log::set_max_level(log_level);
+}
+
+#[cfg(feature = "test")]
+pub mod logging_tests {
+    use crate::{count_failures, set_log_level_or_default};
+    use log::*;
+    use std::{env, panic};
+
+    pub fn run_tests() {
+        println!();
+        let mut failures = 0;
+
+        count_failures!(failures, {
+            test_log_level();
+            test_log_default_greater_than_max();
+        });
+
+        if failures != 0 {
+            panic!("{}: {} tests failed", file!(), failures);
+        }
+    }
+
+    fn test_log_level() {
+        env::set_var("LOG_LEVEL", "WARN");
+        set_log_level_or_default(LevelFilter::Error, LevelFilter::Info);
+        assert_eq!(log::max_level(), LevelFilter::Warn);
+        info!("If you see this, logging is not working correctly!"); // This is not ideal, but checking stdout will be an overkill
+
```

Comment:
Is it easy to maybe implment your own logger that collects log messages to a `Vec<String>`?
You can then check its contents

Reply:
That's a good idea, but the point is to check **our** implementation of the logger. Having a custom logger impl. for tests is not ideal imo

Reply:
Yeah, but this test check the behaviour of the logging system and whether or not it routes messages correctly, not the specific logger, so for these specific purposes, I think it's ok.

Reply:
Yeah thats a fair point